Source/WTF/wtf/Assertions.cpp

```diff
@@ -124,10 +124,18 @@ void disableForwardingVPrintfStdErrToOSLog()
 
 extern "C" {
 
+#if PLATFORM(COCOA)
+static os_log_t webkitSubsystemForGenericOSLog()
+{
+    static os_log_t subsystem = os_log_create(LOG_CHANNEL_WEBKIT_SUBSYSTEM, "Generic");
```

Comment:
We compile with `-Wno-threadsafe-statics` so I think this will require `pthread_once`  or `std::call_once` or similar machinery to work properly with multiple threads in the case that multiple threads happen to call `WTFLogAlways` at about the same time.

(I think for various reasons `pthread_once` actually calls down to `_os_once` while `std::call_once` doesn't so in theory it is probably slightly preferred here.)

Reply:
Actually since you are guarding with `PLATFORM(COCOA)` you can just use dispatch_once with a block.

Reply:
Thanks, good catch! I'll  use `std::call_once` since this is a cpp file and I have no idea how to turn a lambda into a block and I really don't want to use function pointers.

Reply:
Woah TIL you can use blocks in cpp files. I'll use `dispatch_once` like you suggested.